libs/adapter/src/lib/event-store-db/event-store-db.event-store.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  EventStoreDBClient,
+  FORWARDS,
+  jsonEvent,
+  JSONEventData,
+  JSONType,
+  NO_STREAM,
+  ResolvedEvent,
+  START,
+} from '@eventstore/db-client';
+import { EventStore, Event } from '@pl-oss/domain-core';
+
+export class EventStoreDBEventStore implements EventStore {
+  constructor(private readonly eventStoreDBClient: EventStoreDBClient) {}
+
+  async append(stream: string, events: Event[], expectedRevision: number): Promise<void> {
+    const jsonEvents = events.map(EventStoreDBEventStore.mapEventToJsonEvent);
+    const determinedRevision = EventStoreDBEventStore.determineExpectedRevision(expectedRevision);
+    await this.eventStoreDBClient
+      .appendToStream(stream, jsonEvents, { expectedRevision: determinedRevision });
+  }
+
+  async read(stream: string): Promise<Event[]> {
+    const direction = FORWARDS;
+    const fromRevision = START;
+    const resolvedEvents = await this.eventStoreDBClient
+      .readStream(stream, { direction, fromRevision });
+    return resolvedEvents.map(EventStoreDBEventStore.mapResolvedEventToEvent);
+  }
+
```

Comment:
Please make all `private static` methods to `public static`, so we can test them independently.

Reply:
Done in https://github.com/points-log/pl-oss/pull/8/commits/9e78bb4629319e3212e6e20d6922d660aed4f26b